site/src/components/Pill/Pill.tsx

```diff
@@ -44,61 +44,71 @@ const themeStyles =
     };
   };
 
-export const Pill: FC<PillProps> = (props) => {
-  const { lightBorder, icon, text = null, type = "neutral", ...attrs } = props;
-  const theme = useTheme();
+export const Pill: FC<PillProps> = forwardRef<HTMLDivElement, PillProps>(
+  (props, ref) => {
+    const {
+      lightBorder,
+      icon,
+      text = null,
+      type = "neutral",
+      ...attrs
+    } = props;
+    const theme = useTheme();
 
-  const typeStyles = useMemo(() => {
-    if (type in themeOverrides) {
-      return themeOverrides[type as keyof typeof themeOverrides](lightBorder);
-    }
-    return themeStyles(type, lightBorder);
-  }, [type, lightBorder]);
+    const typeStyles = useMemo(() => {
```

Comment:
Does this need to be memoized? I guess it just feels like the logic wouldn't have much overhead, and it's only being used within the same component

Reply:
I don't think so, I guess it is only using `memo` as a way to wrap/encapsulate it into a function but I'm just guessing.